### PR TITLE
feat: add install with go support, fix flags

### LINKED ISSUE #45: provide an option or specific version name to download cosign via go install command
**Description**

in [setup-ko](https://github.com/imjasonh/setup-ko) project there is a special version name called `tip` which allows us to install ko via go install from main. It'd be helpful to people who might want to work with the latest changes of the `cosign` before waiting for its newer version.

### PR DESCRIPTION
**Pull request by developer-guy**
#### Summary

feat: add install with go support, fix flags

#### Ticket Link

Fixes #45 

#### Release Note

```release-note
feat: add install with go support, fix flags
```

### CODE REVIEW

**Review by cpanato — CHANGES_REQUESTED** [review]
looks cool!
can you add some doc in the readme?

**Comment by developer-guy** [comment]
> looks cool! can you add some doc in the readme?
@dentrax and @necatican can you please help me with this folks? ✌️

File: action.yml [hunk 1]
```diff
@@ -33,6 +33,12 @@ runs:
           alias log_error="echo \"ERROR:\""
         fi
         set -e
+        
+        if [[ ${{ inputs.cosign-release }} == "main" ]]; then
+          echo "INFO: installing cosign via go install from its main version"
```
Comment (bobcallaway) [inline action.yml:38]:
```suggestion
          log_info "installing cosign via 'go install' from its main version"
```

Comment (developer-guy) [reply action.yml:38]: thanks for the tip @bobcallaway ✌️

**Comment by developer-guy** [comment]
> looks cool! can you add some doc in the readme?

We(w/@dentrax @necatican)'ve completed the documentation of our new feature. Thank you @cpanato ✌️

File: action.yml [hunk 2]
```diff
@@ -33,6 +33,12 @@ runs:
           alias log_error="echo \"ERROR:\""
         fi
         set -e
+        
+        if [[ ${{ inputs.cosign-release }} == "main" ]]; then
+          log_info "installing cosign via 'go install' from its main version"
+          go install github.com/sigstore/cosign/cmd/cosign@main
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply action.yml:39]: Ah, yes! Thank you, @bobcallaway, for bringing it to my attention  🙋🏻‍♂️ Do you recommend adding this to the information message?

[the remark this answers is not in the document]
Comment (bobcallaway) [reply action.yml:39]: why not just set `GOBIN=${{ inputs.install-dir }}` ahead of the `go install` command, or copy/symlink it after its done?

[the remark this answers is not in the document]
Comment (developer-guy) [reply action.yml:42]: amazing idea @bobcallaway, thank you for bringing it to my attention. I've updated the GA according to your feedback.

File: action.yml [hunk 3]
```diff
@@ -32,6 +32,17 @@ runs:
...
         set -e
+        
+        mkdir -p ${{ inputs.install-dir }}
+
+        if [[ ${{ inputs.cosign-release }} == "main" ]]; then
+          log_info "installing cosign via 'go install' from its main version"
+          GOBIN=$(go env GOPATH)/bin
+          go install github.com/sigstore/cosign/cmd/cosign@main
+          ln -s $GOBIN/cosign ${{ inputs.install-dir}}/cosign
+          echo "${{ inputs.install-dir }}" >> $GITHUB_PATH
```
Comment (bobcallaway) [inline action.yml:43]:
```suggestion
```
this is done below in a platform-independent way so i think you can remove this line

**Review by bobcallaway — COMMENTED** [review]
one comment otherwise LGTM

**Review by bobcallaway — APPROVED** [review]
thanks @developer-guy :)

**Review by cpanato — APPROVED** [review]

**Pull request merged**